tests/CollectorTest.php

```diff
@@ -14,27 +14,566 @@
+		$this->assertSame('fourth', $c->last()->get('path'));
+
+		//Test when "WhereIn" gets single match with duplicate in options
+		$c = new Collector(self::getArray());
+		$c->whereIn('path', ['first', 'first']);
+
+		$this->assertEquals(1, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->first());
+		$this->assertSame('first', $c->first()->get('path'));
+
+		//Test when "WhereIn" gets multiple matches with duplicate in options
+		$c = new Collector(self::getArray());
+		$c->whereIn('path', ['second', 'third', 'third', 'fourth']);
+
+		$this->assertEquals(3, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->first());
+		$this->assertSame('second', $c->first()->get('path'));
+
+		$this->assertInstanceOf(PublicView::class, $c->last());
+		$this->assertSame('fourth', $c->last()->get('path'));
+	}
+
+	public function test_where_not_in_between(): void {
+		//Test when "whereNotIn" gets single match
+		$c = new Collector(self::getArray());
+		$c->whereNotIn('path', ['first', 'second', 'fourth', 'fifth', 'sixth']);
+
+		$this->assertEquals(1, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->first());
+		$this->assertSame('third', $c->first()->get('path'));
+
+		//Test when "WhereNotIn" gets no data
+		$c = new Collector(self::getArray());
+		$c->whereNotIn('path', ['first', 'second', 'third', 'fourth', 'fifth', 'sixth']);
+
+		$this->assertEquals(0, $c->count());
+
+		//Test when "WhereNotIn" gets multiple match
+		$c = new Collector(self::getArray());
+		$c->whereNotIn('path', ['first', 'third', 'fifth', 'sixth']);
+
+		$this->assertEquals(2, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->first());
+		$this->assertSame('second', $c->first()->get('path'));
+
+		$this->assertInstanceOf(PublicView::class, $c->last());
+		$this->assertSame('fourth', $c->last()->get('path'));
+
+		//Test when "WhereNotIn" gets single match with duplicate in options
+		$c = new Collector(self::getArray());
+		$c->whereNotIn('path', ['first', 'first', 'second', 'fourth', 'fifth', 'sixth']);
+
+		$this->assertEquals(1, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->first());
+		$this->assertSame('third', $c->first()->get('path'));
+
+		//Test when "WhereNotIn" gets multiple matches with duplicate in options
+		$c = new Collector(self::getArray());
+		$c->whereNotIn('path', ['first', 'first', 'third', 'fifth', 'sixth']);
+
+		$this->assertEquals(2, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->first());
+		$this->assertSame('second', $c->first()->get('path'));
+
+		$this->assertInstanceOf(PublicView::class, $c->last());
+		$this->assertSame('fourth', $c->last()->get('path'));
+	}
+
+	public function test_order_by(): void {
+		//Test "OrderBy" string case-insensitive
+		$c = new Collector(self::getNotOrderedArray());
+		$c->orderBy('path', false);
+
+		$this->assertEquals(5, $c->count());
+		$this->assertInstanceOf(PublicView::class, $c->get()[0]);
```

Comment:
I feel like all these "assertInstanceOf" in this function would be better in a foreach of `$c->get()`... (also in test_order_by_desc())